src/app/pages/student/details/tabs/student-details-hours/student-details-hours.component.ts

```diff
@@ -69,15 +69,19 @@ export class StudentDetailsHoursComponent implements OnInit {
       if (!data) {
         return;
       }
+      const onlyCustomizedSlots: TimeSlot[] = this.student.schedule.filter((slot) => slot.customized);
+      const newCustomizedSlot: TimeSlot = { index: data.index, lesson: data.lesson, location: data.location, customized: true };
+      const newCustomizedSchedule = [...onlyCustomizedSlots, newCustomizedSlot];
```

Comment:
merging the new customized slot in, without duplicates, and without class originated data.